web/gridt/tests/integration/resources/test_user.py

```diff
@@ -23,9 +30,214 @@ def test_update_bio(self):
 
     def test_update_bio_improperly(self):
         with self.app_context():
-            movement = self.create_movement()
-            user = self.create_user_in_movement(movement, generate_bio=True)
+            user = self.create_user(generate_bio=True)
 
             resp = self.request_as_user(self.users[0], "PUT", "/bio", json={})
 
             self.assertEqual(resp.get_json(), {"message": "Bad request."})
+
+    def test_change_password_with_no_password(self):
+        with self.app_context():
+            user = self.create_user()
+
+            resp = self.request_as_user(
+                self.users[0], "POST", "/change_password", json={}
+            )
+
+            self.assertIn("message", resp.get_json())
+            self.assertEqual(resp.status_code, 400)
+
+            resp = self.request_as_user(
+                self.users[0],
+                "POST",
+                "/change_password",
+                json={"old_password": self.users[0]["password"]},
+            )
+
+            self.assertIn("message", resp.get_json())
+            self.assertEqual(resp.status_code, 400)
+
+    def test_change_password_with_wrong_password(self):
+        with self.app_context():
+            user = self.create_user()
+
+            resp = self.request_as_user(
+                self.users[0],
+                "POST",
+                "/change_password",
+                json={
+                    "old_password": "gibberish",
+                    "new_password": "somethingyoullneverguess",
+                },
+            )
+
+            self.assertIn("message", resp.get_json())
+            self.assertEqual(resp.status_code, 400)
+
+    def test_change_password_with_correct_password(self):
+        # json should contain old_password (self.users[0]["password"]) and new password
```

Comment:
You can give this text as a second input in the function. Then this text will show up when the test fails.
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertEqual

Reply:
Maybe this is the wrong position to write that text.

Reply:
Which lines does this refer to?

Reply:
Line 78